src/hash_map.h

```diff
@@ -50,7 +50,22 @@ unsigned int MurmurHash2(const void* key, size_t len) {
   return h;
 }
 
-#ifdef _MSC_VER
+#if (__cplusplus >= 201103L) || (_MSC_VER >= 1900)
+#include <unordered_map>
+
+namespace std {
+template<>
+struct hash<StringPiece> {
+  typedef StringPiece argument_type;
+  typedef std::size_t result_type;
```

Comment:
I think it's OK for the ``operator()`` to take the parameter by ``const&``. I basically copy-pasted the custom ``std::hash`` from http://en.cppreference.com/w/cpp/utility/hash.